src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisDataFetcher.java

```
@@ -169,6 +193,11 @@ void advanceIteratorTo(String sequenceNumber, InitialPositionInStreamExtended in
             nextIterator = getIterator(ShardIteratorType.AT_SEQUENCE_NUMBER.toString(), sequenceNumber);
         }
         if (nextIterator == null) {
+            final Set<String> childShardIds = leaseCoordinator.getChildShardIds(shardId);
+            if (CollectionUtils.isNullOrEmpty(childShardIds)) {
+                throw new IllegalStateException("Failed to advance iterator for shard " + shardId
```

Looking into this again, we should log error and fail silently. Else when the customer migrates from previous version, they would be stuck in initialization.

---

Synced offline, will not check childShardIds because no matter whether they exist, we will shutdown the shardConsumer